Client/Impl/CustomHeaderInterceptor.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Grpc.Core;
+using Grpc.Core.Interceptors;
+
+namespace Zeebe.Client.Impl
+{
+    /// <summary>
+    ///     Intercept outgoing call to inject metadata
+    ///     The "user-agent" is already filled by grpc-dotnet
+    ///     typically something like: key=user-agent, value=grpc-dotnet/2.53.0 (.NET 7.0.5; CLR 7.0.5; net7.0; windows; x64)
+    /// </summary>
+    internal class CustomHeaderInterceptor : Interceptor
```

Comment:
:question: Was this extracted? Why the change of the user agent?

Reply:
I wanted to maintain the behavior that you have implemented with a specific user-agent description. If I don't do this you will receive on the server, the generic description of a .net grpc agent.